astronomy/frames.hpp

```diff
@@ -19,57 +19,131 @@ using quantities::si::ArcMinute;
 using quantities::si::ArcSecond;
 using quantities::si::Degree;
 
-// A reference frame with a basis.
-// The frame is the International Celestial Reference Frame.
-// The basis is defined from the orbit of the Earth at J2000.0 as follows:
-// The xy plane is the plane of the Earth's orbit at J2000.0.
-// The x axis is out along the ascending node of the instantaneous plane of the
-// Earth's orbit and the Earth's mean equator at J2000.0.
-// The z axis is perpendicular to the xy-plane in the directional (+ or -) sense
-// of Earth's north pole at J2000.0.
-// The basis is right-handed and orthonormal.
-using ICRFJ2000Ecliptic =
-    Frame<serialization::Frame::SolarSystemTag,
-          serialization::Frame::ICRF_J2000_ECLIPTIC, true>;
+// The International Celestial Reference System.
+// The origin is the barycentre of the solar system.  The axes are fixed with
+// respect to the quasars.
+// The xy plane is close to the mean equator at J2000.0.
+// The x axis is close to the dynamical equinox at J2000.
+// The z axis is perpendicular to the xy plane and points to the northern side
+// of the xy plane.
+// To be honest:
+// - There are several realizations of the ICRS, namely ICRF (ICRF1) and its
+//   extensions, ICRF2, and the optical Hipparcos Celestial Reference Frame
+//   (HCRF); these realizations have different errors, but are consistent. ICRF3
+//   is in preparation, as well as an optical realization based on Gaia data.
+// - The terms Barycentric Celestial Reference System (BCRS) and ICRS are often
+//   used interchangeably; the relativistic framework, including the metric
```

Comment:
Should we have a typename for BCRS?

Reply:
No, the question is whether this should be called BCRS instead of ICRS.